doc/frameworks/tensorflow/upgrade_from_legacy.rst

```diff
@@ -0,0 +1,254 @@
+######################################
+Upgrade from Legacy TensorFlow Support
+######################################
+
+With v2 of the SageMaker Python SDK, support for legacy SageMaker TensorFlow images has been deprecated.
+This guide explains how to upgrade your SageMaker Python SDK usage.
+
+For more information about using TensorFlow with the SageMaker Python SDK, see `Use TensorFlow with the SageMaker Python SDK <using_tf.html>`_.
+
+.. contents::
+
+********************************************
+What Constitutes "Legacy TensorFlow Support"
+********************************************
+
+This guide is relevant if one of the following applies:
+
+#. You are using TensorFlow versions 1.4-1.10
+#. You are using TensorFlow versions 1.11-1.12 with Python 2, and
+
+   - you do *not* have ``script_mode=True`` when creating your estimator
+   - you are using ``sagemaker.tensorflow.model.TensorFlowModel`` and/or ``sagemaker.tensorflow.model.TensorFlowPredictor``
+
+#. You are using a pre-built SageMaker image whose URI looks like ``520713654638.dkr.ecr.<region>.amazonaws.com/sagemaker-tensorflow:<tag>``
+
+If one of the above applies, then keep reading.
+
+**************
+How to Upgrade
+**************
+
+We recommend that you use the latest supported version of TensorFlow because that's where we focus our development efforts.
+For information about supported versions of TensorFlow, see the `AWS documentation <https://aws.amazon.com/releasenotes/available-deep-learning-containers-images>`_.
+
+For general information about using TensorFlow with the SageMaker Python SDK, see `Use TensorFlow with the SageMaker Python SDK <using_tf.html>`_.
+
+Training Script
+===============
+
+Newer versions of TensorFlow require your training script to be runnable as a command-line script, similar to what you might run outside of SageMaker. For more information, including how to adapt a locally-runnable script, see `Prepare a Training Script <using_tf.html#id1>`_.
+
+In addition, your training script needs to save your model. If you have your own ``serving_input_fn`` implementation, then that can be passed to an exporter:
+
+.. code:: python
+
+    import tensorflow as tf
+
+    exporter = tf.estimator.LatestExporter("Servo", serving_input_receiver_fn=serving_input_fn)
+
+For an example of how to repackage your legacy TensorFlow training script for use with a newer version of TensorFlow,
+see `this example notebook <https://github.com/awslabs/amazon-sagemaker-examples/blob/master/sagemaker-python-sdk/tensorflow_moving_from_framework_mode_to_script_mode/tensorflow_moving_from_framework_mode_to_script_mode.ipynb>`_.
+
+Inference Script
+================
+
+Newer versions of TensorFlow Serving require a different format for the inference script. Some key differences:
```

Comment:
https://github.com/aws/sagemaker-python-sdk/blob/v1.19.0/src/sagemaker/tensorflow/deploying_python.rst#endpoint-prediction-request-handling implies that only `input_fn` and `output_fn` can be overridden (this is a different interface from what MXNet/PyTorch/etc. use).

With the [current TFS interface](https://github.com/aws/sagemaker-tensorflow-serving-container#prepost-processing), there is an option for implementing the inference function by overriding all of the default behavior in one fell swoop via `handler()` (similar to `transform_fn()` for MXNet), but I didn't think it was relevant here because a user coming from the old TF serving interface wouldn't have an equivalent already.